### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by JacobCutts13**

### CODE REVIEW

File: cypress/integration/homePage.spec.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+describe('Test if the app visits the homepage', () => {
```
Comment (christian-2009) [inline cypress/integration/homePage.spec.js:1]: Would've been nice to have more tests

File: src/components/main.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import TopTen from "./topten";
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+interface TopDogsInterface {
+  breed_name: string;
+  count: number;
+}
+
+export default function Main(): JSX.Element {
```
Comment (christian-2009) [inline src/components/main.tsx:12]: capitalised the function name but not the file name. Try keep it consistent

File: src/components/main.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  // topDogs is an array that stores the breed_Names and vote count for the top10 dogs
```
Comment (christian-2009) [inline src/components/main.tsx:13]: nice use of comments. Makes it really easy to follow :)

File: src/components/main.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  const [topDogs, setTopDogs] = useState<TopDogsInterface[]>([]);
+  // toggleRefresh sets the refresh state to be false to prevent a refresh of the next render
+  const [toggleRefresh, setToggleRefresh] = useState<boolean>(false);
+
+  // GET request to pull the top10 dogs from our database
+  useEffect(() => {
+    async function getTopDogs() {
+      const response = await axios.get(
+        "https://emmalaja-pupvote.herokuapp.com/votes"
```
Comment (christian-2009) [inline src/components/main.tsx:22]: Maybe put the website name in a variable which will make the code more readable and allow you to point to local server when in development (with some extra lines of code added in)

File: src/components/topthree.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  async function fetchImage3() {
+    const name_url = nameToImageURL(top3Dogs[2].breed_name);
+    const URL_STRING =
+      "https://dog.ceo/api/breed/" + name_url + "/images/random";
+    const response = await axios.get(URL_STRING);
+    setDogImage3(response.data.message);
+  }
+
+  //
```
Comment (christian-2009) [inline src/components/topthree.tsx:62]: Nice job splitting up the functionality. Very clean and easy to read. Good structure

File: src/components/topthree.tsx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  useEffect(() => {
+    fetchImage3();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // takes dogscore and translates it into pawPrints
+  function getPawsArray(score: number): string[] {
+    let count = 0;
+    const arr = [];
+    while (count < score) {
```
Comment (christian-2009) [inline src/components/topthree.tsx:82]: naming of arr can be abit too general, but good comment.

File: src/components/topthree.tsx [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      count++;
+    }
+    return arr;
+  }
+
+  const dogPawArray1 = getPawsArray(dog1Score);
+  const dogPawArray2 = getPawsArray(dog2Score);
+  const dogPawArray3 = getPawsArray(dog3Score);
+
+  // returns three cards with breed_name, score, image, paw-vote ratio
```
Comment (christian-2009) [inline src/components/topthree.tsx:93]: amazing comments

File: src/components/vote.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  const dogbreed1 = dogbreedfinder(dog1);
+  const dogbreed2 = dogbreedfinder(dog2);
+
+  // POST request to submit vote for specific dog breed
+  async function submitFavourites(dogbreed: string) {
+    await axios.post("https://emmalaja-pupvote.herokuapp.com/", {
+      breed_name: dogbreed,
+    });
+    playSound();
+    setSubmitted((previous) => !previous);
```
Comment (christian-2009) [inline src/components/vote.tsx:44]: nice use of useState arrow function

File: src/components/topten.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      const newTopTenFullInfo = await Promise.all(unresolvedMappedTopDogs);
+      // We use the top3 Dogs to cover the refresh time of the carousel
+      const topThreeFullInfo = newTopTenFullInfo.slice(0, 7);
+      // We are initialising the state topTenFullInfo
+      setTopTenFullInfo([...newTopTenFullInfo, ...topThreeFullInfo]);
+    };
+    getAllImageURL();
+  }, [props.top10Dogs]);
+
+  // We are mapping our top10 Dogs into our TopDogFullInfoCard custom component
```
Comment (christian-2009) [inline src/components/topten.tsx:83]: very clean

File: src/utils/dogbreedfinder.test.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    dogbreedfinder(
+      "https://images.dog.ceo/breeds/terrier-lakeland/n02095570_1283.jpg"
+    )
+  ).toBe("terrier-lakeland");
+  expect(
+    dogbreedfinder(
+      "https://images.dog.ceo/breeds/terrier-american/n02093428_4939.jpg"
+    )
+  ).toBe("terrier-american");
+});
```
Comment (christian-2009) [inline src/utils/dogbreedfinder.test.ts:19]: good tests but more edge cases could be better

File: src/components/TopDogFullInfoCard.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+export default function TopDogFullInfoCard(
+  props: topDogFullInfoCardProps
+): JSX.Element {
+  // Props used to map dogs from top10dogs individually
+  return (
+    <>
+      <section className="top10card">
+        <h2>{props.breed_name}</h2>
+        <div className="top10card--info">
+          <h3>{(props.position % 10) + 1}</h3>
```
Comment (christian-2009) [inline src/components/TopDogFullInfoCard.tsx:17]: some comments to better understand what's going on here

File: src/utils/nameToImageURL.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+/*
+breed_name = breed(-sub_breed)
+image_url = breed(/sub_breed)
+*/
+
+export default function nameToImageURL(breed_name: string): string {
+  const breed_name_lowerCase = breed_name.toLowerCase();
```
Comment (NobeenIslam) [inline src/utils/nameToImageURL.ts:7]: great clear naming. Camel case might be cool

File: src/utils/readNameURLFormatter.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
+// Format dog names to correct format without impacting the database or API fetches ( .toUpperCase cannot be applied to a null value )
+export function readNameURLFormatter(url: string): string {
+  let read_name = "";
```
Comment (NobeenIslam) [inline src/utils/readNameURLFormatter.ts:3]: Just some casing on variables :). But clear logic

**Review by christian-2009 — COMMENTED** [review]
amazing comments. Really great splitting up the components and it was really nice to read. Rename files though to actually match the component names.

File: cypress/integration/1-getting-started/todo.spec.js [hunk 14]
```diff
@@ -0,0 +1,143 @@
+/// <reference types="cypress" />
```
Comment (NobeenIslam) [inline cypress/integration/1-getting-started/todo.spec.js:1]: Might be worth removing examples for code reviews. (A lot of files! XD)
